chain/polkadot/keys_test.go

```diff
@@ -0,0 +1,86 @@
+package polkadot_test
+
+import (
+	"crypto/ed25519"
+	"encoding/hex"
+	"testing"
+
+	p2pCrypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/strangelove-ventures/ibctest/chain/polkadot"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNodeKeyPeerID(t *testing.T) {
+	nodeKey, err := hex.DecodeString("1b57e31ddf03e39c58207dfcb5445958924b818c08c303a91838e68cfac551b2")
+	require.NoError(t, err, "error decoding node key from hex string")
+
+	privKeyEd25519 := ed25519.NewKeyFromSeed(nodeKey)
+	privKey, _, err := p2pCrypto.KeyPairFromStdKey(&privKeyEd25519)
+	require.NoError(t, err, "error getting private key")
+
+	id, err := peer.IDFromPrivateKey(privKey)
+	require.NoError(t, err, "error getting peer id from private key")
+	peerId := peer.Encode(id)
+	require.Equal(t, "12D3KooWCqDbuUHRNWPAuHpVnzZGCkkMwgEx7Xd6xgszqtVpH56c", peerId)
+}
+
+func Test_DeriveEd25519FromName(t *testing.T) {
+	privKey, err := polkadot.DeriveEd25519FromName("Alice")
+	require.NoError(t, err, "error deriving ed25519 private key")
+
+	pubKey, err := privKey.GetPublic().Raw()
+	require.NoError(t, err, "error fetching pubkey bytes")
+
+	pubKeyEncoded, err := polkadot.EncodeAddressSS58(pubKey)
+	require.NoError(t, err, "error encoding ed25519 public key to ss58")
+
+	require.Equal(t, "5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu", pubKeyEncoded)
+}
+
+func Test_DeriveSr25519FromNameAccount(t *testing.T) {
+	privKeyAccount, err := polkadot.DeriveSr25519FromName([]string{"Alice"})
+	require.NoError(t, err, "error deriving account sr25519 private key")
+
+	pubKeyAccount := make([]byte, 32)
+	for i, mkByte := range privKeyAccount.Public().Encode() {
+		pubKeyAccount[i] = mkByte
+	}
```

Comment:
I assume this was written in this style due to an error like `invalid operation: C() (value of type [4]byte) (slice of unaddressable value)`.

Instead of the byte-by-byte copy, put the array in a temporary variable and then slice it:

```go
b := privKeyAccount.Public().Encode()
pubKeyAccount := b[:]
// Or if there is a chance that the array b may change, copy it instead:
pubKeyAccount = append([]byte(nil), b[:]...)
```

Reply:
done